### PR TITLE
feat(ast, ast_codegen): pass the `scope_id` to the `enter_scope` event.

### PR DESCRIPTION
**Pull request by rzvxa**

### CODE REVIEW

**Comment by rzvxa** [comment]
* **#4168** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4168?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rzvxa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_isolated_declarations/src/scope.rs [hunk 1]
```diff
@@ -291,8 +294,9 @@ impl<'a> Visit<'a> for ScopeTree<'a> {
     ///             ^^^^^^^^^^^^^^^^^^^^
     /// We need to add both `T` and `K` to the scope
     fn visit_ts_mapped_type(&mut self, ty: &TSMappedType<'a>) {
+        // TODO: doesn't have a scope_id!
+        self.enter_scope(ScopeFlags::empty(), &Cell::default());
```
Comment (rzvxa) [inline crates/oxc_isolated_declarations/src/scope.rs:298]: What are we going to do about these?

Comment (rzvxa) [reply crates/oxc_isolated_declarations/src/scope.rs:298]: @Dunqing

Comment (Dunqing) [reply crates/oxc_isolated_declarations/src/scope.rs:298]: Leave it for me. I will fully refactor it later

**Comment by rzvxa** [comment]
@overlookmotel If you are good with this PR go ahead and merge it in.

**Review by Dunqing — APPROVED** [review]

**Review by overlookmotel — APPROVED** [review]

**Pull request merged**